numba/np/arraymath.py

```diff
@@ -4159,6 +4159,27 @@ def np_select_arr_impl(condlist, choicelist, default=0):
 
     return np_select_arr_impl
 
+
+@overload(np.asarray_chkfinite)
+def np_asarray_chkfinite(a, dtype=None, order='C'):
+
+    msg = "The argument to np.asarray_chkfinite must be array-like"
+    if not isinstance(a, (types.Array, types.Sequence, types.Tuple)):
+        raise TypingError(msg)
+
+    if is_nonelike(dtype):
+        dt = a.dtype
+    else:
+        dt = dtype.dtype
+
+    def impl(a, dtype=None, order='C'):
+        a = np.asarray(a, dtype=dt)
+        if not np.all(np.isfinite(a)):
+            raise TypingError("array must not contain infs or NaNs")
```

Comment:
There are two comments regarding this exception. Firstly, it'll be best to raise a `ValueError` here as this is what Numpy does and additionally this code is no longer in a "typing context" so a `TypingError` makes little sense. Also, please do check, if you haven't already done so, if the error message is the same as in Numpy ( or confirm that this is the case.)


https://numpy.org/doc/stable/reference/generated/numpy.asarray_chkfinite.html

Reply:
I have changed TypingError to ValueError. The error message here is same as in Numpy.